src/resolve/resolved-link-bus.c

```diff
@@ -61,6 +61,15 @@ static int property_get_dns_internal(
         if (r < 0)
                 return r;
 
+        LIST_FOREACH(servers, s, l->manager->dns_servers) {
+                if (dns_server_ifindex(s) != l->ifindex)
+                        continue;
+
+                r = bus_dns_server_append(reply, s, false, extended);
+                if (r < 0)
+                        return r;
+        }
```

Comment:
hmm, waaa? this doesn't look right? the per-link scopes never use the global DNS servers, this shouldn't be exposed here... i figure the "resolvectl dns" client side should query the global DNS servers too and show them somehow, but it shouldn't be the dbus server side that mangles that...

Reply:
Hmm, but manager's DNS=/DNSEx= bus property mixed servers assigned to Manager and Links. So, it is hard to dedup in the client side... WDYT about introduce ManagerDNS= or GlobalDNS= bus property that only lists servers assigned to Manager? Then, resolvectl uses it.

Reply:
yeah, makes sense